airbyte-webapp/src/packages/cloud/views/DefaultView.tsx

```diff
@@ -28,10 +28,11 @@ export const DefaultView: React.FC = () => {
       }
       replace
       // exp-signup-selected-source-definition
-      {...(isSignupSourceSelectorExperiment && {
-        state: { sourceDefinitionId },
-        to: `/${RoutePaths.Workspaces}/${workspaces[0].workspaceId}/${RoutePaths.Connections}/${RoutePaths.ConnectionNew}`,
-      })}
+      {...(isSignupSourceSelectorExperiment &&
+        sourceDefinitionId && {
+          state: { sourceDefinitionId },
+          to: `/${RoutePaths.Workspaces}/${workspaces[0].workspaceId}/${RoutePaths.Connections}/${RoutePaths.ConnectionNew}`,
+        })}
```

Comment:
Maybe this could be cleaned up to make it easier to follow the logic? A better way might be to set up the props before the render. For example:

```tsx
let navigateProps: NavigateProps = { replace: true }
if (somecondition) {
  navigateProps = {
     ...navigateProps,
     state: { sourceDefintionId },
     to: `...`
  }
} else {
   navigateProps.to = `...`
}

return <Navigate {...navigateProps} />
```

Reply:
The reason we keep it separate is that the experiments are meant to be removed completely potentially when they are not successful, so the code is tried to be highlighted separately. Leti, treat this as a 🧹 that needs to be cleaned up if we keep this.

Reply:
Of course!